src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,491 @@
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetUseCompensatedSums",
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (onoff)
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNTRUE);
+    ark_mem->step = sprkStep_TakeStep_Compensated;
+  }
+  else
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNFALSE);
+    ark_mem->step = sprkStep_TakeStep;
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethod:
+
+  Specifies the SPRK method
+
+  ** Note in documentation that this should not be called along
+  with SPRKStepSetOrder. **
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethod(void* arkode_mem, ARKodeSPRKStorage sprk_storage)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethod", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = sprk_storage;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethodName:
+
+  Specifies the SPRK method.
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethodName(void* arkode_mem, const char* method)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethodName", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = ARKodeSPRKStorage_LoadByName(method);
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetOrder:
+
+  Specifies the method order
+
+  ** Note in documentation that this should not be called along
+  with SPRKStepSetMethod. **
+  ---------------------------------------------------------------*/
+int SPRKStepSetOrder(void* arkode_mem, int ord)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetOrder", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  /* set user-provided value, or default, depending on argument */
+  if (ord <= 0) { step_mem->q = 4; }
+  else { step_mem->q = ord; }
```

Comment:
Should this also check for an upper bound, and/or whether `ord` matches "allowable" positive values?